test/e2e-go/upgrades/application_support_test.go

```diff
@@ -0,0 +1,550 @@
+	primaryNodeUnupgradedProtocol := consensus[consensusTestFastUpgrade(protocol.ConsensusFuture)]
+	primaryNodeUnupgradedProtocol.ApprovedUpgrades = make(map[protocol.ConsensusVersion]uint64)
+	primaryNodeUnupgradedProtocol.ApprovedUpgrades[consensusTestFastUpgrade(protocol.ConsensusFuture)] = 0
+	consensus[consensusTestUnupgradedProtocol] = primaryNodeUnupgradedProtocol
+
+	client := fixture.GetLibGoalClientForNamedNode("Primary")
+	secondary := fixture.GetLibGoalClientForNamedNode("Node")
+	err := config.SaveConfigurableConsensus(client.DataDir(), consensus)
+	require.NoError(t, err)
+
+	fixture.Start()
+
+	defer fixture.Shutdown()
+
+	accountList, err := fixture.GetNodeWalletsSortedByBalance(client.DataDir())
+	require.NoError(t, err)
+
+	creator := accountList[0].Address
+	wh, err := client.GetUnencryptedWalletHandle()
+	require.NoError(t, err)
+
+	user, err := client.GenerateAddress(wh)
+	require.NoError(t, err)
+
+	fee := uint64(1000)
+
+	round, err := client.CurrentRound()
+	require.NoError(t, err)
+
+	// Fund the manager, so it can issue transactions later on
+	_, err = client.SendPaymentFromUnencryptedWallet(creator, user, fee, 10000000000, nil)
+	require.NoError(t, err)
+	client.WaitForRound(round + 2)
+
+	round, err = client.CurrentRound()
+	require.NoError(t, err)
+
+	// There should be no apps to start with
+	ad, err := client.AccountData(creator)
+	require.NoError(t, err)
+	require.Zero(t, len(ad.AppParams))
+
+	ad, err = client.AccountData(user)
+	require.NoError(t, err)
+	require.Zero(t, len(ad.AppParams))
+	require.Equal(t, basics.MicroAlgos{Raw: 10000000000}, ad.MicroAlgos)
+
+	counter := `#pragma version 2
+// a simple global and local calls counter app
+byte b64 Y291bnRlcg== // counter
+dup
+app_global_get
+int 1
++
+app_global_put  // update the counter
+int 0
+int 0
+app_opted_in
+bnz opted_in
+err
+opted_in:
+int 0  // account idx for app_local_put
+byte b64 Y291bnRlcg== // counter
+int 0
+byte b64 Y291bnRlcg==
+app_local_get
+int 1  // increment
++
+app_local_put
+int 1
+`
+	approval, err := logic.AssembleString(counter)
+	require.NoError(t, err)
+	clearstate, err := logic.AssembleString("int 1")
+	require.NoError(t, err)
+	schema := basics.StateSchema{
+		NumUint: 1,
+	}
+
+	// create the app
+	tx, err := client.MakeUnsignedAppCreateTx(
+		transactions.OptInOC, approval, clearstate, schema, schema, nil, nil, nil,
+	)
+	require.NoError(t, err)
+	tx, err = client.FillUnsignedTxTemplate(creator, round, round+primaryNodeUnupgradedProtocol.DefaultUpgradeWaitRounds, fee, tx)
+	require.NoError(t, err)
+	signedTxn, err := client.SignTransactionWithWallet(wh, nil, tx)
+	require.NoError(t, err)
+	round, err = client.CurrentRound()
+	require.NoError(t, err)
+
+	_, err = client.BroadcastTransaction(signedTxn)
+	require.NoError(t, err)
+
+	// this transaction is expect to reach the first node ( primary ), but to be rejected by the second node when transmitted over gossip.
```

Comment:
Where are we checking that the transaction is rejected by the second node?

Reply:
we check the transaction pool of the second node : 
```
	// check that the secondary node doesn't have that transaction in it's transaction pool.
	pendingTx, err = secondary.GetPendingTransactions(1)
	require.NoError(t, err)
	require.Equal(t, uint64(0), pendingTx.TotalTxns)
```

Reply:
Even if it is accepted, it won't be in the transaction pool forever right?

Reply:
correct. it would remain there until `lastRound` passes. Keep in mind that once a transaction is included in a transaction pool, it would remain there until it's no longer valid ( i.e. passed `lastRound` ) or included in a block.
We know that it wasn't included in a block since we tested both nodes, and we know that blocks are always uniform.